File: engine/src/P7m.Engine.Core/SharedMemory/SharedMemoryPath.cs
namespace P7m.Engine.Core.SharedMemory;

/// <summary>
/// Resolve o nome lógico do mapa (<c>sharedMemoryMapName</c>) para o arquivo
/// físico, com a mesma regra do escritor Node.js
/// (middleware/src/sharedmem/SharedMemoryPath.ts).
/// </summary>
public static class SharedMemoryPath
{
    public static string Resolve(string mapName)
    {
        var runtimeDir = Environment.GetEnvironmentVariable("XDG_RUNTIME_DIR");
        var baseDir = !string.IsNullOrEmpty(runtimeDir) && !OperatingSystem.IsWindows()
            ? runtimeDir
            : Path.GetTempPath();
        return Path.Combine(baseDir, $"{mapName}.mmap");
    }
}

File: middleware/src/sharedmem/SharedMemoryPath.ts
import os from "node:os";
import path from "node:path";

/**
 * Resolve o nome lógico do mapa (`sharedMemoryMapName`) para o arquivo físico,
 * com a mesma regra do leitor C# (engine/src/P7m.Engine.Core/SharedMemory/SharedMemoryPath.cs).
 */
export function resolveSharedMemoryPath(mapName: string): string {
  const runtimeDir = process.platform !== "win32" ? process.env["XDG_RUNTIME_DIR"] : undefined;
  const baseDir = runtimeDir && runtimeDir.length > 0 ? runtimeDir : os.tmpdir();
  return path.join(baseDir, `${mapName}.mmap`);
}
